cli/internal/graph/graph.go

```diff
@@ -76,12 +78,19 @@ func (g *CompleteGraph) GetPackageTaskVisitor(
 			return fmt.Errorf("Could not find definition for task")
 		}
 
+		// Task env mode is only independent when global env mode is `infer`.
+		taskEnvMode := globalEnvMode
+		if taskEnvMode == util.Infer && taskDefinition.PassthroughEnv != nil {
+			taskEnvMode = util.Strict
+		}
+
 		// TODO: maybe we can remove this PackageTask struct at some point
 		packageTask := &nodes.PackageTask{
 			TaskID:          taskID,
 			Task:            taskName,
 			PackageName:     packageName,
 			Pkg:             pkg,
+			EnvMode:         taskEnvMode,
```

Comment:
Is this used off `PackageTask` anywhere? If it's just for dry json / run summary, we might not need it here

Reply:
We need it for the task hash itself in case globalEnvMode is `infer`: https://github.com/vercel/turbo/pull/4529/files#diff-7790aa2d704a01d8951c2bdc87cdf80f13b234ba1b4db29cd1502e7b84e4d3e4R287